File: 2021/day01.nim
import strutils
import sequtils
import sugar

let test1 = "199,200,208,210,200,207,240,269,260,263".split(",").mapIt(parseInt(it))
let integers = readFile("day01.txt").splitLines().mapIt(parseInt(it))

proc sol1(integers: seq[int]): int =
    var cur: int = integers[0]
    for i in 1..<integers.len():
        if integers[i] > cur:
            result += 1
        cur = integers[i]

proc sol2(integers: seq[int]): int =
    var cur: int = foldl(integers[0..2], a+b)
    var s: int = 0
    for i in 1..<integers.len()-2:
        s = foldl(integers[i..i+2], a+b)
        if s > cur:
            result += 1
        cur = s


assert sol1(test1) == 7
assert sol2(test1) == 5
dump sol1(integers)
dump sol2(integers)

File: 2021/day01.py
def sol1(data):
    sol = 0
    cur = data[0]
    for e in data[1:]:
        if e > cur:
            sol += 1
        cur = e
    return sol

def sol2(data):
    sol = 0
    cur = sum(data[0:3])
    tmp = 0
    for i in range(1, len(data)-2):
        s = sum(data[i:i+3])
        if s > cur:
            sol += 1
        cur = s
    return sol



if __name__ == "__main__":
    data = [int(e.strip()) for e in open('day01.txt', 'r').readlines()]
    print(sol1(data))
    print(sol2(data))
